canvas_sdk/utils/http.py

```diff
@@ -0,0 +1,55 @@
+import time
+from functools import wraps
+from typing import Any, Callable, TypeVar
+
+import requests
+import statsd
+
+F = TypeVar("F", bound=Callable)
+
+
+class Http:
+    """A helper class for completing HTTP calls with metrics tracking."""
+
+    def __init__(self) -> None:
+        self.session = requests.Session()
+        self.statsd_client = statsd.StatsClient()
+
+    @staticmethod
+    def measure_time(fn: F | None = None) -> Callable[[F], F] | F:
+        """A decorator to store timing of HTTP calls."""
+
+        def _decorator(fn: F) -> F:
+            @wraps(fn)
+            def wrapper(self: "Http", *args: Any, **kwargs: Any) -> Any:
+                print(fn.__name__)
+                start_time = time.time()
+                result = fn(self, *args, **kwargs)
+                end_time = time.time()
+                timing = int((end_time - start_time) * 1000)
+                self.statsd_client.timing(f"http_{fn.__name__}", timing)
+                return result
+
+            return wrapper
+
+        return _decorator(fn) if fn else _decorator
+
+    @measure_time
+    def get(self, url: str, headers: dict = {}) -> requests.Response:
```

Comment:
A nice usability enhancement for all of these would be to allow passing in an optional dictionary of query parameters that we could encode for them, rather than making the user do it each time. This would map to the `params` argument in requests.